### ISSUE TITLE
LATERAL JOINS don't work on distributed hypertables

### ISSUE THREAD

**Issue by svenklemm** [timescale/timescaledb#3128]
Example Query:
```
SELECT * FROM generate_series(1,3) g(device) 
JOIN LATERAL (SELECT time FROM t WHERE t.device = g.device LIMIT 10) l ON true;
```
Error message:
```
ERROR:  [data_node_1]: bind message supplies 0 parameters, but prepared statement "" requires 1
```

**Comment by jgm-ktg** [comment]
@svenklemm Can you provide the definition/data of `t` above and how the distributed hypertable was partitioned?  As mentioned on #3094, I was able to seemingly successfully run a lateral join on a distributed hypertable.  In my case, only the EXPLAIN failed with an error similar to above.

**Comment by svenklemm** [comment]
I think this was fixed as part of #4176
